lib/zendesk_apps_tools/api_connection.rb

```diff
@@ -1,11 +1,26 @@
 module ZendeskAppsTools
   module APIConnection
-    FULL_URL     = /https?:\/\//
-    URL_TEMPLATE = 'https://%s.zendesk.com/'
+    DEFAULT_URL_TEMPLATE = 'https://%s.zendesk.com/'
+    # taken from zendesk/lib/vars.rb
+    SUBDOMAIN_VALIDATION_PATTERN = /^[a-z0-9][a-z0-9\-]{1,}[a-z0-9]$/i
+    ZENDESK_URL_VALIDATION_PATTERN = /^(https?):\/\/[a-z0-9]+(([\.]|[\-]{1,2})[a-z0-9]+)*\.([a-z]{2,16}|[0-9]{1,3})((:[0-9]{1,5})?(\/?|\/.*))?$/ix
```

Comment:
As discussed, consider making the protocol group optional e.g.
`/^((https?):\/\/)?[a-z0-9]+(([\.]|[\-]{1,2})[a-z0-9]+)*\.([a-z]{2,16}|[0-9]{1,3})((:[0-9]{1,5})?(\/?|\/.*))?$/ix`

And prepend `https://` when it isn't present but when the input otherwise looks like a complete URL

Reply:
Actually on second thought, this might be really difficult to do as to determine whether it looks like a complete URL you'd need to validate that it ends in a valid TLD. The regex for that might be pretty long and prone to change. I'll leave it up to you to determine whether this would be a worthwhile change

Reply:
Cool. I'll go with the current regular expression for now.

The regex for that would be longer and more complex. We will need to prepend https or http in front of the input string. I don't think it is impossible.

Reply:
I guess this can also be confusing. What is someone enters a url without protocol, then do we assume it to be https? or http? Because by default browsers will resolve it to http, but in our case we would want to resolve it to https in most cases.